server/pulp/server/content/web/views.py

```diff
@@ -137,10 +137,13 @@ def get(self, request):
         host = request.get_host()
         path = os.path.realpath(request.path_info)
 
-        # Authorization
-        if not allow_access(request.environ, host):
-            # Not Authorized
-            return HttpResponseForbidden()
+        # Check authorization if http isn't being used. This environ variable must
+        # be available in all implementations so it is not dependant on Apache httpd:
+        # https://www.python.org/dev/peps/pep-0333/#environ-variables
+        if request.environ['wsgi.url_scheme'] != 'http':
```

Comment:
What will stop users from accessing https content by substituting http:// on the URL? It doesn't seem that self.redirect() checks whether the content should be https protected or not. Does something else in the chain check this?